tests/compiler/builtins.ts

```diff
@@ -39,6 +39,13 @@ assert(!isFunction(changetype<u32>(0)));
 assert(isNullable(changetype<C | null>(0)));
 assert(!isNullable(changetype<C>(0)));
 
+// compile time check
+export function foo(a:i32[]): void {
+  if (isReference(a[0])) {
+    ERROR();
+  }
```

Comment:
I guess it will be much better use `!isReference(a[0])` here due to this `ERROR` will throw in any case (compile and non-compile) for now

Reply:
Sorry I don't get point. You mean it should be a string array and check !isReference?

Reply:
It would be more correct to check fixtures for absence of codegen than for its presence. This is how you can implicitly check how much tree-shrinkage works for this case. `a[0]` is not reference, so `isReference(a[0])` should eliminate branch with ERROR in debug and release wats. Isn't that the purpose of this PR?

Reply:
Ah, right! It should be eliminate branch with `ERROR`. It's just a little confusing that we have quite a lot of code in release

Reply:
a[0] has side effect, and a lot of shadow stack operator. Maybe it is a good point to optimize assemblyscript performance in complex project.